components/FilterForm/index.js

```diff
@@ -1,67 +1,93 @@
 import styled from "styled-components";
 import React, { useState } from "react";
-import { plants } from "@/lib/data";
 
-export default function FilterForm({ onFilterUpdate }) {
+export default function FilterForm({ onFilterUpdate, plants }) {
   const [plantSize, setPlantSize] = useState("");
-  const [sunlightRequirement, setsunlightRequirement] = useState("");
-  const [waterNeeds, setwaterNeeds] = useState("");
-  const [optimalTemperature, setoptimalTemperature] = useState("");
-  const [petFriendly, setpetFriendly] = useState("");
+  const [sunlightRequirement, setSunlightRequirement] = useState("");
+  const [waterNeeds, setWaterNeeds] = useState("");
+  const [optimalTemperature, setOptimalTemperature] = useState("");
+  const [petFriendly, setPetFriendly] = useState("");
 
   const handleFilter = (event) => {
     event.preventDefault();
 
-    const newFilteredPlants = plants.filter(plant => 
-      (plantSize ? plant.size === plantSize : true) &&
-      (sunlightRequirement ? plant.sunlightRequirements === sunlightRequirement : true) &&
-      (waterNeeds ? plant.waterNeeds === waterNeeds : true) &&
-      (optimalTemperature ? plant.optimalTemperature === optimalTemperature : true) &&
-      (petFriendly ? plant.petFriendly === (petFriendly === "true") : true)
-      );
+    const newFilteredPlants = plants.filter(
+      (plant) =>
+        (plantSize ? plant.size === plantSize : true) &&
+        (sunlightRequirement
+          ? plant.sunlightRequirements === sunlightRequirement
+          : true) &&
+        (waterNeeds ? plant.waterNeeds === waterNeeds : true) &&
+        (optimalTemperature
+          ? plant.optimalTemperature === optimalTemperature
+          : true) &&
+        (petFriendly ? plant.petFriendly === (petFriendly === "true") : true)
+    );
```

Comment:
This seems challenging to comprehend, but if you want to refine the logic in the future, I would suggest you use small, simple functions for each filter comparison, and stitch them together afterwards:
```
// very simplified
if (filterPlantSize(arg1, arg2)
    && filterSunlightRequirement(arg1, arg2)
    …
)
```
There are several approaches to this, but it works nevertheless!